### ISSUE TITLE
Cannot read property 'toUpperCase' of undefined

### ISSUE THREAD

**Issue by zhangsheng** [rackt/react-router#2162]
when I use this with webpack, it always throw this error:

Uncaught TypeError: Cannot read property 'toUpperCase' of undefined

**Comment by subjectix** [comment]
That error happens when [`React.createClass`](https://facebook.github.io/react/docs/top-level-api.html#react.createclass) has null or undefined as its argument. This happens to JSX elements too since they are transformed to `React.createClass` calls. Somehow, somewhere a require call does not resolve and returns undefined (from what I guess). you should do a divide-and-conquer test to find which files causes this, track down the specific component that ends up undefined and see why the require resolution fails :cold_sweat:

**Comment by ubill** [comment]
I'm not sure if this part solves the error:

"correct i.e. working code"
```
 <Provider store={store}>
    {() =>
      <Router history={history}>
        <Route component={MainContainer}>
          <Route path="/" component={App} />
          <Route path="/:itemuuid" component={AppTemp} />
        </Route>
      </Router>
    }
  </Provider>
```

It won't work if without the anonymous function:
```
 <Provider store={store}>
    
      <Router history={history}>
        <Route component={MainContainer}>
          <Route path="/" component={App} />
          <Route path="/:itemuuid" component={AppTemp} />
        </Route>
      </Router>
    
  </Provider>
```

I am not quite sure about this, I just remember seeing it an hour ago (but I didn't write down what was the root cause after I moved on from this) when I was moving code about when trying to solve the issue before this. Try this and see whether it solves your problem in the immediate term.

**Comment by knowbody** [comment]
@subjectix thanks for the answer, again the issue is not related to React Router. you must be doing something wrong on the way. hard to point out what without any code.

@ubill you are using redux which and the solution is not related to this issue, at least redux was not mentioned here.

**Comment by ubill** [comment]
@knowbody & @subjectix You are right the problem mentioned here most likely has nothing to do with react-router (so too the possible solution that I posted, not surprisingly) 

I just noticed now the original question did not mention about react-router too. I just wanted to help him if possible since I recently saw the same error message.

Cheers and thanks for the clarification! It does look like nothing to do with react-router but more of not being able to resolve the "JSX Component transformation" elsewhere in the asker's code.
